# Bivalves as emerging model systems to study the mechanisms and evolution of sex determination: a genomic point of view â€” Data and code

In this repository you will find data and codes used to perform the analyses of *Dmrt* genes in bivalve molluscs for the perspective:
> Nicolini F, [Ghiselli F](https://github.com/fghiselli), [Luchetti A](https://github.com/andluche), Milani L. (2023). Bivalves as emerging model systems to study the mechanisms and evolution of sex determination: a genomic point of view. *Genome Biology and Evolution*, evad181. [https://doi.org/10.1093/gbe/evad181](https://doi.org/10.1093/gbe/evad181)
> 
> **Abstract.** Bivalves are a diverse group of molluscs that have recently attained a central role in plenty of biological research fields, thanks to their peculiar life history traits. Here we propose that bivalves should be considered as emerging model systems also in sex-determination studies, since they would allow to investigate: (i) the transition between environmental and genetic sex determination, with respect to different reproductive backgrounds and sexual systems (from species with strict gonochorism to species with various forms of hermaphroditism); (ii) the genomic evolution of sex chromosomes, considering that no heteromorphic sex chromosomes are currently known and that homomorphic sex chromosomes have been identified just in few species of scallops; (iii) the putative role of mitochondria at some level of the sex determination signaling pathway, in a mechanism that may resemble the cytoplasmatic male sterility of plants; (iv) the evolutionary history of sex-determination related gene families with respect to other animal groups. In particular, we think that this last topic may lay the foundations for expanding our understanding of bivalve sex determination, as our current knowledge is quite fragmented and limited to few species. As a matter of fact, tracing the phylogenetic history and diversity of sex-determination related gene families (such as the *Dmrt*, *Sox* and *Fox* genes) would allow to perform more targeted functional experiments and genomic analyses, but also fostering the possibility of establishing a solid comparative framework.

<p align="center">
<img src="./images/fig_2.png", height="500">
</p>

Feel free to visit our group GitHub page **[CompBio-BO](https://github.com/CompBio-BO)**.
  
## Infos
  - [<code>ML_phylogeny/</code>](./ML_phylogeny/) contains the data that are necessary to infer the ML tree in [**Fig. 2A**](./images/fig_2.);
  - [<code>compilation_boxplot/</code>](./compilation_boxplot/) contains the data that are necessary to generate plots in [**Fig. 2B-C**](./images/fig_2.);
  - [<code>assembly_dataset.tsv</code>](./assembly_dataset.tsv) is the online version of the **Table 1** from the manuscript, so that it can be easily parsed;
  - [<code>dmrt_gene_IDs.tsv</code>](./dmrt_gene_IDs.tsv) is a table with accession numbers and identities of the sequences used in the ML phylogenetic analysis.
